ocrd_network/ocrd_network/processing_server.py

```diff
@@ -0,0 +1,289 @@
+            return self._processor_list
+        res = set([])
+        for host in self.config.hosts:
+            for processor in host.processors:
+                res.add(processor.name)
+        self._processor_list = list(res)
+        return self._processor_list
+
+    async def push_processor_job(self, processor_name: str, data: JobInput) -> JobOutput:
+        """ Queue a processor job
+        """
+        if processor_name not in self.processor_list:
+            raise HTTPException(
+                status_code=status.HTTP_404_NOT_FOUND,
+                detail='Processor not available'
+            )
+
+        # validate additional parameters
+        if data.parameters:
+            ocrd_tool = get_ocrd_tool_json(processor_name)
+            if not ocrd_tool:
+                raise HTTPException(
+                    status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+                    detail=(f'Processor \'{processor_name}\' not available. It\'s ocrd_tool is '
+                            'empty or missing')
+                )
+            validator = ParameterValidator(ocrd_tool)
+            report = validator.validate(data.parameters)
+            if not report.is_valid:
+                raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail=report.errors)
+
+        # determine path to mets if workspace_id is provided
+        if bool(data.path) == bool(data.workspace_id):
+            raise HTTPException(
+                status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
+                detail='Either \'path\' or \'workspace_id\' must be set'
+            )
+        elif data.workspace_id:
+            workspace = await Workspace.get(data.workspace_id)
+            if not workspace:
+                raise HTTPException(
+                    status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
+                    detail=f'Workspace for id \'{data.workspace_id}\' not existing'
+                )
+            data.path = workspace.workspace_mets_path
+
+        job = Job(**data.dict(exclude_unset=True, exclude_none=True), processor_name=processor_name,
+                  state=StateEnum.queued)
+        await job.insert()
+        processing_message = OcrdProcessingMessage.from_job(job)
+        encoded_processing_message = OcrdProcessingMessage.encode_yml(processing_message)
+        if self.rmq_publisher:
+            self.rmq_publisher.publish_to_queue(processor_name, encoded_processing_message)
+        else:
+            raise Exception('RMQPublisher is not connected')
+        return job.to_job_output()
+
+    async def get_processor_info(self, processor_name) -> Dict:
+        """ Return a processor's ocrd-tool.json
+        """
+        if processor_name not in self.processor_list:
+            raise HTTPException(
+                status_code=status.HTTP_404_NOT_FOUND,
+                detail='Processor not available'
+            )
+        return get_ocrd_tool_json(processor_name)
```

Comment:
This implementation requires that all supported processors must be installed on the same machine with Processing Server as well, which might not be the case. Maybe after integrating [#884](https://github.com/OCR-D/core/pull/884), we can send requests to each processor to ask for its information instead.

Reply:
I concur – see [earlier discussion](https://github.com/OCR-D/core/pull/974#discussion_r1094282394) above.

Reply:
what's the status here, cf. https://github.com/OCR-D/core/pull/974#discussion_r1094282394

Reply:
> Maybe after integrating #884, we can send requests to each processor to ask for its information instead.

The processing worker is not a server anymore to send requests there. I still have no clear idea how to achieve that. The best idea I have found so far is to store the ocrd tool jsons in the DB so the Processing Server can retrieve the information from there.

Reply:
to be continued [here](https://github.com/OCR-D/core/issues/1034)